java/core/src/main/java/co/worklytics/psoxy/utils/email/EmailAddress.java

```diff
@@ -0,0 +1,21 @@
+package co.worklytics.psoxy.utils.email;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class EmailAddress {
```

Comment:
our own structure, in lieu of javax / jakarta `InternetAddress` stuff; i do think this possibly removes/changes some implicit IP/domain validation ... but don't care so much for our use-case